internal/generator/generator.go

```diff
@@ -0,0 +1,92 @@
+package generator
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/nestoca/joy-generator/internal/gitrepo"
+	"github.com/nestoca/joy/api/v1alpha1"
+	"github.com/nestoca/joy/pkg/catalog"
+)
+
+type Generator struct {
+	repo *gitrepo.GitRepo
+}
+
+type Result struct {
+	// Release holds the release's values loaded from the yaml file in the catalog
+	Release *v1alpha1.Release `json:"release"`
+
+	// Environment holds the environment info where the release will be deployed. The full spec is not loaded to minimize the payload size
+	Environment *Environment `json:"environment"`
+
+	// RenderedValues is a yaml string that is the Release.spec.values rendered with any templated fields
+	RenderedValues string `json:"templatedValues"`
```

Comment:
Do we really need to be explicit in the name that values are templated?  I would much, much prefer simply `values`.  Support for templating is a feature, it is really secondary in terms of concept.  When consuming those values in the AppSet, we should simply think of them as "the values". And there will not be values other than the rendered ones.  Also "templatedValues" seems ambiguous as to whether or not the templating has been performed at that point or if the values just "contain templating".  Anyway, can we please just name that `values`? 🙏

Seems like we often argue on similar cases, where you insist on very technically explicit names, whereas I prefer the simpler term, even if it might not convey the full entire thorough holistic semantic of what the thing is! 😉

Reply:
I named it as such because I wanted to have a clear distinction between the values loaded from the release, which is an object, and the rendered values which is a yaml string. This makes the template in the ApplicationSet easier to read and understand. 

> Also "templatedValues" seems ambiguous as to whether or not the templating has been performed at that point or if the values just "contain templating".

Yeah thats why I changed it to `renderedValues`, just forgot to update the json tag.
```suggestion
	RenderedValues string `json:"renderedValues"`
```

Reply:
I see your point, I'm just not convince we need to emphasize that distinction that much.  Are you worried because theoretically, the template could access both from the result?  Like `.Result.release.spec.values` and `.Result.renderedValues`? And you don't want those two things to be confused?  Because honestly I would not worry about that, the former is quite deep in the tree and it would fail anyway it they tried to use it instead.  I'm still more in favour of `values`, for the sake of simplicity. But for the sake of good neighbourhood, if you are really, really, really against it, I could live with `renderedValues` (not saying I will sleep very well, but at least I will live).